lib/internal/vm/Module.js

```diff
@@ -0,0 +1,150 @@
+'use strict';
+
+const { isContext } = process.binding('contextify');
+const errors = require('internal/errors');
+const util = require('util');
+
+const {
+  ModuleWrap,
+  kUninstantiated,
+  kInstantiating,
+  kInstantiated,
+  kEvaluating,
+  kEvaluated,
+  kErrored,
+} = internalBinding('module_wrap');
+
+const StatusMap = {
+  [kUninstantiated]: 'uninstantiated',
+  [kInstantiating]: 'instantiating',
+  [kInstantiated]: 'instantiated',
+  [kEvaluating]: 'evaluating',
+  [kEvaluated]: 'evaluated',
+  [kErrored]: 'errored',
+};
+
+let globalModuleId = 0;
+const kWrap = Symbol('wrap');
+const kContextModuleCount = Symbol('contextModuleCount');
+const kDependencyCache = Symbol('dependencyCache');
+
+class Module {
+  constructor(src, options = {}) {
+    let context;
+    if (options.context !== undefined && isContext(options.context.sandbox)) {
+      context = options.context.sandbox;
+      if (options.context[kContextModuleCount] === undefined)
+        throw new errors.TypeError('ERR_CONTEXT_NOT_MODULE_CONTEXT');
+    }
+
+    const url = options.url ||
+      `vm:module(${context !== undefined ?
+        options.context[kContextModuleCount]++ :
+        globalModuleId++})`;
+
+    const wrap = new ModuleWrap(src, url, {
+      context,
+      lineOffset: options.lineOffset,
+      columnOffset: options.columnOffset,
+    });
+
+    Object.defineProperties(this, {
+      [kWrap]: { value: wrap },
```

Comment:
ModuleWrap should not be accessible to user code, use a WeakMap until private fields land.

Reply:
it isn't accessible in user code in any way, the worst that can happen is a user seeing the object, and they can see it on github anyway

Reply:
Being able to see the symbol means being able to access the values:

```mjs
o = {[Symbol('kWrap')]:{private:()=>{throw 'leaked'}}}; o[Object.getOwnPropertySymbols(o)[0]].private()
```

ModuleWrap is an internal binding and should not leak. The value is accessible.